Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -0,0 +1,691 @@
+    write-host $([Constants]::DoubleDashLine)
+     
+    Write-Host "[step 1 of 4]: Checking for prerequisites." 
+
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..."
+            Setup-Prerequisites
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+    }
+        
+    # Connect to AzAccount
+    $isContextSet = Get-AzContext
+    if ([string]::IsNullOrEmpty($isContextSet))
+    {       
+        Write-Host "Connecting to AzAccount..."
+        Connect-AzAccount -ErrorAction Stop
+        Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting context for current subscription.
+    $currentSub = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop -Force
+    Write-Host "Metadata Details: `nSubscriptionId: $($SubscriptionId) `nAccountName: $($currentSub.Account.Id) `nAccountType: $($currentSub.Account.Type)" 
+    Write-Host $([Constants]::SingleDashLine) 
+    Write-Host "Starting with subscription [$($SubscriptionId)]..." 
+    write-host $([Constants]::DoubleDashLine)
+
+    write-host "[step 2 of 4]: Validating the Account type."
+
+    Write-Host " ***To perform rollback operation for disabling secure transfer user must have atleast contributor access on Storage Account(s) of subscription: [$($SubscriptionId)]  ***" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host "Validating whether the current user [$($currentSub.Account.Id)] have valid account type [User] to run the script for subscription [$($SubscriptionId)]..." 
+
+    # Safe Check: Checking whether the current account is of type User 
+    if($currentSub.Account.Type -ne "User")
+    {
+        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor $([Constants]::MessageType.Warning)
+        break;
+    }
+
+    Write-Host "Successfully validated" -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine) 
+    Write-Host "Fetching remediation log to perform rollback operation on  Storage Account(s) from subscription [$($SubscriptionId)]..." 
+    write-host $([Constants]::DoubleDashLine)
+
+    # Array to store Storage Accounts.
+    $storageAccounts = @()
+
+    if (-not (Test-Path -Path $FilePath))
+    {
+        Write-Host "Error: Control file path is not found." -ForegroundColor $([Constants]::MessageType.Error)
+        break;        
+    }
+   
+    # Fetching remediated log 
+    Write-Host "[step 3 of 4 ]: Getting the list of resources for rollback."
+    $remediatedResourceLog = Import-Csv -LiteralPath $FilePath
+    
+    Write-Host "Fetching remediation log..."
+    $resource =@()
```

Comment:
Is `resource` being used anywhere in the roll back code?

Reply:
no. Removed it.